reports/3_generate_report_emails.py

```diff
@@ -0,0 +1,144 @@
+DEV_LINK_BASE = f'https://development-build--cal-itp-reports.netlify.app/gtfs_schedule/{PUBLISH_DATE_YEAR}/{PUBLISH_DATE_MONTH}'    
+
+
+# +
+df_crm_emails = pd.read_csv(
+    "https://docs.google.com/spreadsheets"
+    "/d/1AHFa5SKcEn7im374mPwULFCj7VlFchQr/export?gid=289636985&format=csv",
+    skiprows=1,
+)
+
+df_crm_emails.columns = df_crm_emails.columns.str.lower()
+
+# +
+tbl.gtfs_schedule.feed_info() >> count(missing_email = _.feed_contact_email.isna())
+
+tbl_report_emails = (
+    tbl.views.reports_gtfs_schedule_index()
+    >> left_join(
+        _,
+        tbl.gtfs_schedule.feed_info()
+        >> select(_.startswith("calitp"), _.feed_contact_email),
+        ["calitp_itp_id", "calitp_url_number"]
+    )
+    # NOTE: reports currently only use url number = 0, so we remove any others --
+    >> filter(_.publish_date == REPORT_PUBLISH_DATE,  _.calitp_url_number == 0)
+    >> select(_.startswith("calitp"), _.agency_name, _.use_for_report, _.feed_contact_email)
+)
+
+report_emails = tbl_report_emails >> collect()
+# -
+
+report_crm_emails = (
+    report_emails
+    >> full_join(
+        _,
+        df_crm_emails
+        >> filter(_.itp_id.notna())
+        >> mutate(itp_id=_.itp_id.astype(int)),
+        {"calitp_itp_id": "itp_id"},
+    )
+)
+
+report_crm_emails_inner = (
+    report_crm_emails
+    >> filter(_.calitp_itp_id.notna(), _.itp_id.notna())
+    >> select(-_.itp_id)
+    >> mutate(calitp_itp_id=_.calitp_itp_id.astype(int))
+)
+
+
+final_all_emails = (
+    report_crm_emails_inner
+    >> transmute(_.calitp_itp_id, _.calitp_url_number, email = _["main email"], origin = "CRM")
+    >> filter(_.email.notna())
+    >> mutate(
+        calitp_itp_id = _.calitp_itp_id.astype(int),
+        calitp_url_number = _.calitp_url_number.astype(int),
+        report_url = REPORT_LINK_BASE + "/" + _.calitp_itp_id.astype(str) + "/",
+        dev_url = DEV_LINK_BASE + "/" + _.calitp_itp_id.astype(str) + "/",
+    )
+)
+
+
+# +
+### send emails using postmarks API
+
+# +
+def _html_body(col):
+    ##return a single formatted html for a single row of the table
+    return f"""<html>
+            <body>Hello!<br> Greetings from the Cal-ITP team.<br> As part of our work with the GTFS feeds for agencies across the state, we prepare a monthly report with some basic statistics and validation results for your agencies' feed.<br>
+            You can view your report for {REPORT_PUBLISH_MONTH} at {col.report_url}. We are actively looking for opportunities to help transit agencies address issues with their GTFS feeds. If you would like to meet with our team to learn more about how we can help, or have any questions, please email our team at hello@calitp.org. <br>
+            <br>Thanks,<br>
+            The Cal-ITP Team<br>
+            https://reports.calitp.org/
+            </body>
+        </html>"""
+
+HTML_MESSAGE = final_all_emails.apply(_html_body, axis =1)
+EMAIL = final_all_emails.email
```

Comment:
Please avoid use of all-caps variable names for variables that are not constants or store dynamically generated data.